portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/blogs/BlogsPermissions.testcase

```diff
@@ -1190,16 +1190,12 @@ definition {
 			userLoginEmailAddress = "userea@liferay.com",
 			userLoginFullName = "userfn userln");
 
-		Navigator.gotoPage(pageName = "Blogs Page");
-
-		BlogsNavigator.gotoEntryPG(
-			entryContent = "Blogs Entry Content",
-			entryTitle = "Blogs Entry Title");
-
-		JSONBlog.addComment(
-			commentBody = "Blog Comment",
+		BlogsNavigator.openToBlogEntry(
 			entryTitle = "Blogs Entry Title",
-			groupName = "Guest");
+			layoutName = "Blogs Page",
+			siteURLKey = "guest");
+
+		Comments.addPG(entryComment = "Blog Comment");
```

Comment:
Why not add a comment through JSON here @Jeremy-Ch ?

Reply:
Hey @austinchiang , If we want to use other users to add comments by JSON, we should update the JSON to change the user as a variable, otherwise, the comment will be added by the Test user whatever you log in to any other users. For this test, we need to add a comment by the new user, so I think that we add it by UI is enough. If you want to use other users to add the comment by JSON, please let me know, I could update it, thanks.
```
@summary = "This calls the JSON WS API to add a Blog entry comment"
	macro addComment {
		var entryId = JSONBlog.getBlogEntryId(
			friendlyURL = "${friendlyURL}",
			entryTitle = "${entryTitle}",
			groupName = "${groupName}");

		var portalURL = JSONCompany.getPortalURL();

		if (!(isSet(parentCommentId))) {
			var parentCommentId = "0";
		}

		var curl = '''
			${portalURL}/o/headless-delivery/v1.0/blog-postings/${entryId}/comments \
			-H 'Content-Type: application/json' \
			-u 'test@liferay.com:test'
			-d '{
				"creator": {},
				"parentCommentId": "${parentCommentId}",
				"text": "${commentBody}"
			}'
		''';

		echo("## Adding a 'Comment' with the following info using JSONWS:");

		echo("## * commentBody: ${commentBody}");

		echo("## * entryId: ${entryId}");

		var messageId = JSONCurlUtil.post("${curl}");
	}
```

Reply:
We have tests that add a comment through the UI so let's refactor and make the JSON macro more flexible. Should be simple, just add a conditional

Reply:
OK, I will send a new pr after updating.